File: sem4/mpp/mpp-proiect-csharp-vreauladudu/persistence/repository/MotorcycleRepository.cs
using mpp_proiect_csharp_vreauladudu.domain;
using System;
using System.Collections.Generic;

namespace mpp_proiect_csharp_vreauladudu.repository
{
    public interface MotorcycleRepository : IRepository<int, Motorcycle>
    {
        List<Motorcycle> findByCapacity(int Capacity);
        List<Motorcycle> findByBrand(String Brand);
        Motorcycle findById(int Id);
    }
}

File: sem4/mpp/mpp-proiect-java-vreauladudu/persistence/src/main/java/moto/repository/MotorcycleRepository.java
package moto.repository;

import moto.Motorcycle;

import java.sql.SQLException;
import java.util.List;

public interface MotorcycleRepository extends IRepository<Integer, Motorcycle>
{
    List<Motorcycle> findByCapacity(Integer Capacity) throws SQLException;
    List<Motorcycle> findByBrand(String Brand) throws SQLException;
    Motorcycle findById(Integer Id) throws SQLException;
}